survey.js

```diff
@@ -60,201 +60,128 @@ class ParticipantInfo {
-    type: jsPsychSurveyHtmlForm,
-    preamble: PREPARE_FOR_SURVEY,
-    html: MULTI_CHOICE_HTML,
-    on_finish : function(data) {
-        if (typeof data.rt === "number") {
-            data.rt = Math.round(data.rt);
-        }
-    }
-};
-
-let survey_review_survey_data = {
-    type: jsPsychHtmlButtonResponse,
-    stimulus: function(data){
-
-        let survey_1 = 
-            jsPsych.data.get().last(2).values()[0].response;
-        
-        let survey_2 = 
-            jsPsych.data.get().last(1).values()[0].response;
-        
-        let b_year      = survey_1.birth_year;
-        let b_month     = survey_1.birth_month;
-        let n_lang      = survey_1.native_language;
-
-        let bilingual   = survey_2.Multilingual;
-        let dyslexic    = survey_2.Dyslexic;
-        let sex         = survey_2.Sex;
-        let hand_pref   = survey_2.HandPreference;
-
-        return `
-            <h1>Your data</h1>
-            <div class='survey'>
-
-            <div><strong>Birth year</strong>: ${b_year} </div>
-            <div><strong>Birth month</strong>: ${b_month} </div>
-            <div><strong>Native language</strong>: ${n_lang} </div>
-            <div><strong>Multilingual</strong>: ${bilingual} </div>
-            <div><strong>Dyslexic</strong>: ${dyslexic} </div>
-            <div><strong>Sex</strong>: ${sex} </div>
-            <div><strong>Hand preference</strong>: ${hand_pref} </div>
-            <br><br>
+}
 
-            <p>Is this information correct?</p>
-            </div>
-            `;
+let survey_2 = {
+    type: IlsSurveyPlugin,
+    fields: {
+        multilingual: {label: 'Multilingual environment', options: {yes: "Yes", no: "No"}},
+        dyslexia: {label: 'Dyslexia', options: {yes: "Yes", no: "No"}},
+	handedness: {label: 'Handedness', options: {right: "Right", left: "Left"}},
+	sex: {label: 'Biological sex', options: {male: "Male", female: "Female", other: "Other", unspecified: "Unspecified"}},
     },
-    choices: [YES_BUTTON_TEXT, NO_BUTTON_TEXT],
-    response_ends_trial: true,
-    on_finish: function(data){
-        // Repeat the survey if yes (0) was not pressed.
-        // this may give multiple entries, up to the researcher to filter out
-        repeat_survey = data.response !== 0;
-        if (typeof data.rt === "number") {
-            data.rt = Math.round(data.rt);
-        }
-    }
-};
+    exclusion: function(data) {
+	// return true when participant should be excluded
+	if (data.dyslexia == 'yes') {
+	    return true;
+	}
+	// accept participant otherwise
+	return false
+    },
+    html: `
+    <div style="text-align: left">
+        <p>Were you born and raised in a multilingual environment?</p>
+        <label><input type="radio" name="multilingual" value="yes" required/>Yes</label>
+        <label><input type="radio" name="multilingual" value="no" required/>No</label>
+
+	<p>Which hand do you prefer to write with?
+            <span class="info-toggle">🛈</span>
```

Comment:
I see encoding/decoding of text/symbols errors here

Reply:
If you want to test the updated version, you have to manually include the dev version of utils (https://web-experiments.lab.hum.uu.nl/jspsych/uil-utils/dev/jspsych-plugins/plugin-survey.js). I'll make a new release later